pkg/frontend/transport/handler_test.go

```diff
@@ -54,12 +55,17 @@ func TestWriteError(t *testing.T) {
 	}{
 		{http.StatusInternalServerError, errors.New("unknown")},
 		{http.StatusGatewayTimeout, context.DeadlineExceeded},
+		{http.StatusGatewayTimeout, errors.Wrap(context.DeadlineExceeded, "an error occurred")},
 		{StatusClientClosedRequest, context.Canceled},
+		{StatusClientClosedRequest, errors.Wrap(context.Canceled, "an error occurred")},
 		{http.StatusBadRequest, httpgrpc.Errorf(http.StatusBadRequest, "")},
+		{http.StatusBadRequest, errors.Wrap(httpgrpc.Errorf(http.StatusBadRequest, ""), "an error occurred")},
+		{http.StatusBadRequest, apierror.New(apierror.TypeBadData, "")},
```

Comment:
Let's test a different status code as well, like apierror.TypeNotFound

Reply:
Done in https://github.com/grafana/mimir/pull/8407/commits/b20987d9904709060472be7339101a8317dfeedf